runtime/src/accounts_index.rs

```diff
@@ -1663,24 +1647,23 @@ impl<T: IndexValue> AccountsIndex<T> {
         );
         let map = &self.account_maps[self.bin_calculator.bin_from_pubkey(pubkey)];
 
-        {
-            let r_account_maps = map.read().unwrap();
-            r_account_maps.upsert(pubkey, new_item, Some(old_slot), reclaims, reclaim);
-        }
+        map.upsert(pubkey, new_item, Some(old_slot), reclaims, reclaim);
         self.update_secondary_indexes(pubkey, account, account_indexes);
     }
 
     pub fn unref_from_storage(&self, pubkey: &Pubkey) {
         let map = &self.account_maps[self.bin_calculator.bin_from_pubkey(pubkey)];
-        map.read().unwrap().unref(pubkey)
+        map.unref(pubkey)
     }
 
     pub fn ref_count_from_storage(&self, pubkey: &Pubkey) -> RefCount {
-        if let Some(locked_entry) = self.get_account_read_entry(pubkey) {
-            locked_entry.ref_count()
-        } else {
-            0
-        }
+        let map = &self.account_maps[self.bin_calculator.bin_from_pubkey(pubkey)];
```

Comment:
same here, can we use the helper function `get_account_maps_read_lock`?

Reply:
done